sdk/src/main/java/co/omise/android/ui/ExpiryDateEditText.kt

```diff
@@ -73,6 +75,15 @@ class ExpiryDateEditText : OmiseEditText {
         override fun onTextChanged(s: CharSequence?, start: Int, before: Int, count: Int) {
             if (s == null || s.length > MAX_CHARS) return
 
+            // Define a regular expression pattern to match non-numeric characters and DATE_SEPARATOR
+            val pattern: Pattern = Pattern.compile("[^0-9" + Pattern.quote(DATE_SEPARATOR) + "]")
+
+            // Use a Matcher to find any non-numeric characters in the string
+            val matcher: Matcher = pattern.matcher(s)
+            if(matcher.find()){
```

Comment:
@AnasNaouchi do we have a way to enforce style here? I noticed a different spacing between `if` and the _expression_.

Reply:
Yes we do have, you can format with android studio, for kotlin files. I was not enabled by default as it would result in some differences but will fix that